File: 700-search-in-a-binary-search-tree/700-search-in-a-binary-search-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */


struct TreeNode* searchBST(struct TreeNode* root, int val){
    struct TreeNode* temp = root;
    
    while(temp){
        if (temp->val == val)
        return temp;
        
        else if(temp->val > val)
        temp = temp->left;
        
        else
        temp = temp->right;
    }
    return NULL;
}

File: 700-search-in-a-binary-search-tree/700-search-in-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def searchBST(self, root, val):
        """
        :type root: TreeNode
        :type val: int
        :rtype: TreeNode
        """
        
        temp = root
        while temp:
            if temp.val==val:
                return temp
            elif temp.val > val:
                temp = temp.left
            else:
                temp = temp.right
        
        return None
        